### ISSUE TITLE
[arm64] Test failed:  JIT_Performance._CodeQuality_BenchmarksGame_mandelbrot_mandelbrot_7_mandelbrot_7_._CodeQuality_BenchmarksGame_mandelbrot_mandelbrot_7_mandelbrot_7_sh

### ISSUE THREAD

**Issue by VincentBu** [dotnet/coreclr#23804]
**Job:**
https://ci.dot.net/job/dotnet_coreclr/job/master/view/arm64/job/jitstress/job/arm64_cross_checked_ubuntu16.04_jitstress2_jitstressregs0x80_tst/163/

**Failed tests:**
JIT_Performance._CodeQuality_BenchmarksGame_mandelbrot_mandelbrot_7_mandelbrot_7_._CodeQuality_BenchmarksGame_mandelbrot_mandelbrot_7_mandelbrot_7_sh

**Log:**
```
JIT_Performance._CodeQuality_BenchmarksGame_mandelbrot_mandelbrot_7_mandelbrot_7_._CodeQuality_BenchmarksGame_mandelbrot_mandelbrot_7_mandelbrot_7_sh [FAIL]
        /n/nReturn code:      1/nRaw output file:      /home/robox/j/workspace/dotnet_coreclr/master/jitstress/arm64_cross_checked_ubuntu16.04_jitstress2_jitstressregs0x80_tst/bin/tests/Linux.arm64.Checked/JIT/Performance/Reports/JIT.Performance/CodeQuality/BenchmarksGame/mandelbrot/mandelbrot-7/mandelbrot-7.output.txt/nRaw output:/nBEGIN EXECUTION
        /home/robox/j/workspace/dotnet_coreclr/master/jitstress/arm64_cross_checked_ubuntu16.04_jitstress2_jitstressregs0x80_tst/bin/tests/Linux.arm64.Checked/Tests/Core_Root/corerun mandelbrot-7.exe ''
        P4
        80 80
        ���������������������������������������������������������������������������������
        ������������������
        ������������������
        �����������������������������������������������������������������������������������������"������������������
        �������������������������������������*���������������*�����������������������������������
        ������������������
        ��������������������������������������������U����������������������������
        ������������������
        ������������������������������������*��������������������*�������������������������������������
        ������������������"���������������������������������������������������������������������������������������������
        ������������������
        ������������������
        ���������������������������������������������������������������������Expected: 100
        Actual: 255
        END EXECUTION - FAILED
        Test Harness Exitcode is : 1
        /nTo run the test:/n> set CORE_ROOT=/home/robox/j/workspace/dotnet_coreclr/master/jitstress/arm64_cross_checked_ubuntu16.04_jitstress2_jitstressregs0x80_tst/bin/tests/Linux.arm64.Checked/Tests/Core_Root/n> /home/robox/j/workspace/dotnet_coreclr/master/jitstress/arm64_cross_checked_ubuntu16.04_jitstress2_jitstressregs0x80_tst/bin/tests/Linux.arm64.Checked/JIT/Performance/CodeQuality/BenchmarksGame/mandelbrot/mandelbrot-7/mandelbrot-7.sh/n
        Expected: True
        Actual:   False
        Stack Trace:
          /mnt/j/workspace/dotnet_coreclr/master/arm64_cross_checked_ubuntu16.04/bin/tests/Linux.arm64.Checked/TestWrappers/JIT.Performance/JIT.Performance.XUnitWrapper.cs(739,0): at JIT_Performance._CodeQuality_BenchmarksGame_mandelbrot_mandelbrot_7_mandelbrot_7_._CodeQuality_BenchmarksGame_mandelbrot_mandelbrot_7_mandelbrot_7_sh()
```

**Comment by AndyAyersMS** [comment]
Assuming codegen for now.

cc @dotnet/jit-contrib

**Comment by sandreenko** [comment]
I was able to repro it on windows x64 with jitstress2_jitstressregs0x80.
There are many other failures with jitstress2_jitstressregs* on x64 in HardwareIntrinsics and SIMD tests.

**Comment by CarolEidt** [comment]
@sandreenko - I believe this is a duplicate of #23861. I see it failing on windows x64 with just jitStressRegs=0x80. Looking at the code for `BenchmarksGame.MandelBrot_7:DoBench`, you can see where it reuses the register that it is keeping the upper half of ymm6 in (V09), 3 instructions prior to using it to restore it:
```
       vextractf128 **ymm9**, ymm6, 1  <--- extracts the upper half of ymm6
       vextractf128 ymm10, ymm7, 1
       vextractf128 ymm11, ymm8, 1
       call     CORINFO_HELP_NEWARR_1_VC
       mov      rdx, 0xD1FFAB1E
       lea      rcx, bword ptr [rax+16]
       vmovdqu  xmm0, qword ptr [rdx]
       vmovdqu  qword ptr [rcx], xmm0
       vmovdqu  xmm0, qword ptr [rdx+16]
       vmovdqu  qword ptr [rcx+16], xmm0
       vmovupd  **ymm9**, ymmword ptr[rax+16] <--- Overwrites it here
       xor      edx, edx
       test     esi, esi
       vinsertf128 ymm6, **ymm9**, 1  <--- restores the upper half with a bad value
```
With #23889 it uses `ymm12` instead of `ymm9` for the `vmovupd`

**Comment by sandreenko** [comment]
@CarolEidt I have tried your change from #23889 and it did not fix this issue (on x64 at least). So I think it needs a separate fix.  But yes they are similar, this issue was caused also by  #23344. Is it too late to revert it?

**Comment by CarolEidt** [comment]
@sandreenko - I ran this on x64 with my fix, and with COMPlus_JitStress=2 and COMPlus_JitStressRegs=0x80 and it passes.

**Comment by sandreenko** [comment]
Could you please try `complus_jitstressregs=0x1000` without jitstress please?

**Comment by sandreenko** [comment]
That is my output (with your change):
```
F:\git\coreclr\bin\tests\Windows_NT.x64.Checked\JIT\Performance\CodeQuality\BenchmarksGame\mandelbrot\mandelbrot-7>set complus
COMPlus_ContinueOnAssert=0
complus_jitstressregs=0x1000
COMPlus_NoGuiOnAssert=0
complus_TieredCompilation=0

F:\git\coreclr\bin\tests\Windows_NT.x64.Checked\JIT\Performance\CodeQuality\BenchmarksGame\mandelbrot\mandelbrot-7>mandelbrot-7.cmd
BEGIN EXECUTION
 "F:\git\coreclr\bin\tests\Windows_NT.x64.Checked\Tests\Core_Root\corerun.exe" mandelbrot-7.exe
P4
80 80
╚êêêêêêêêê√╗╗╗╗╗╗╗║ê                    ╚êêêêêêêêê╚êêêêêêêêê                  ╚êêêêêêêêê                         0  ╚êêêêêêêêê                31  ╚êêêïï╗╗êê╚êêêêêêêêê          ╚êêêï╗╗╗╗ê╚êêêë╗╗╗╗ê╚êêêêêêêêê          ╚êêê╗╗╗╗╗ê╚êêê╗╗╗╗╗ê╚êêêêêêêêê          ╚êêê╗╗╗╗╗╕╚êêê╗╗╗╗╗╕╚êêêêêêêêê          ╚ï╗ë╗╗╗╗╗╕╚ï╗ê╗╗╗╗╗╕╚êêêêêêêêê          ╚¢╗╣╗╗╗╗╗╕╚ï╗╣╗╗╗╗╗¿╚êêêêêêêêê          ╦¢╗╗╗╗╗╗╗ê╦╗╗╗╗╗╗╗╗ê╚êêêêêêêêê          ╦¢╗╗╗╗╗╗╗ê╚¢╗╣╗╗╗╗╗¿╚êêêêêêêêê          ╚¢╗╣╗╗╗╗╗╕╚ï╗╣╗╗╗╗╗╕╚êêêêêêêêê          ╚ï╗ë╗╗╗╗╗╕╚èêë╗╗╗╗╗¿╚êêêêêêêêê          ╚êêê╗╗╗╗╗╕╚êêê╗╗╗╗╗ê╚êêêêêêêêê          ╚êêê╗╗╗╗╗ê╚êêêï╗╗╗╗ê╚êêêêêêêêê          ╚êêêï╗╗╗╗ê╚êêêï¢╗╗╣ê╚êêêêêêêêê                31        "  ╚êêêêêêêêê                         #   ╚êêêêêêêêê                            ╚êêêêêêêêê                              Expected: 100
Actual: -1
END EXECUTION - FAILED
FAILED
```

**Comment by sandreenko** [comment]
and if you add `complus_jitstress=2` that you get #23904:
```
F:\git\coreclr\bin\tests\Windows_NT.x64.Checked\JIT\Performance\CodeQuality\BenchmarksGame\mandelbrot\mandelbrot-7>mandelbrot-7.cmd
BEGIN EXECUTION
 "F:\git\coreclr\bin\tests\Windows_NT.x64.Checked\Tests\Core_Root\corerun.exe" mandelbrot-7.exe
FailFast:
Infinite recursion during resource lookup within System.Private.CoreLib.  This may be a bug in System.Private.CoreLib, or potentially in certain extensibility points such as assembly resolve events or CultureInfo names.  Resource name: NotSupported_WrongResourceReader_Type
   at System.Environment.FailFast(System.String)
   at System.SR.InternalGetResourceString(System.String)
   at System.SR.GetResourceString(System.String, System.String)
   at System.Resources.ResourceReader._ReadResources()
   at System.Resources.ResourceReader.ReadResources()
   at System.Resources.ResourceReader..ctor(System.IO.Stream)
   at System.RuntimeMethodHandle.InvokeMethod(System.Object, System.Object[], System.Signature, Boolean, Boolean)
   at System.Reflection.RuntimeConstructorInfo.Invoke(System.Reflection.BindingFlags, System.Reflection.Binder, System.Object[], System.Globalization.CultureInfo)
```

**Comment by CarolEidt** [comment]
The issue that shows up with jitStressRegs=0x1000 appears to be an encoding issue with `vinsertf128`. We think we're generating:
```
vinsertf128 ymm6, ymmword ptr[V09+0x10 rsp+90H], 1
```
but what we get is:
```
vinsertf128 ymm6,ymm0,xmmword ptr [rsp+90h],1
```
Where the `ymm0` should be `ymm6` to get the two-operand semantic that we want.

**Comment by tannergooding** [comment]
@CarolEidt, this is an issue with the Numerics Vector code, correct?

I think we print all operands and test this encoding path for the HWIntrinsics codegen, but wanted to double check...

**Comment by CarolEidt** [comment]
Sort of ... this impacts both System.Numerics.Vector code as well has HWIntrinsics vector code, as it's the code that saves & restores the upper half of large vectors around calls.

**Comment by tannergooding** [comment]
Sorry, to clarify, I was specifically referring to the insertf128 encoding issue, which sounds like it is separate?

I believe the non-hwintrinsic codegen for the simd instructions often goes down a different code path that doesn't always handle the vex encoding directly (and also causes other side effects like the disasm not displaying the operand)

**Comment by CarolEidt** [comment]
Yes, this is not directly related to the encodings, but to the use of the fake 2-operand form of 3-operand instructions (which were used even more extensively before we had as much emitter support).

**Comment by CarolEidt** [comment]
I've added the fix for this to #23889

**Comment by BruceForstall** [comment]
@CarolEidt This test is still failing on Linux/arm64 for R2R GCStress=F runs. I opened https://github.com/dotnet/coreclr/issues/24224 to track, in case it's a different issue.
